### PR TITLE
Disable request diagnostics modal on install

### PR DESCRIPTION
**Pull request by markmhendrickson**
> Try out Leather build e6cd821 — [Extension build](https://github.com/leather-wallet/extension/actions/runs/8830430964), [Test report](https://leather-wallet.github.io/playwright-reports/feat/remove-request-diagnostics), [Storybook](https://feat-remove-request-diagnostics--65982789c7e2278518f189e7.chromatic.com), [Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=feat/remove-request-diagnostics)

This PR removes the "Help is improve" dialog from display upon installation of the extension. This change is intended to reduce friction during onboarding.

We may add the ability to disable pseudonymous analytics later inside of the app's general settings and outside of the onboarding flow.

## Before

image

## After

image

### CODE REVIEW

File: src/app/features/container/utils/route-helpers.ts
Comment (pete-watters) [inline src/app/features/container/utils/route-helpers.ts]:
```suggestion
  return pathname.match(RouteUrls.Onboarding); // need to match get-started/ledger
```

I can clean this up further as I will refactor this area

File: src/app/routes/app-routes.tsx
Comment (pete-watters) [inline src/app/routes/app-routes.tsx:132]: You need to remove the import of `AllowDiagnosticsModal` and also delete that file `@app/pages/onboarding/allow-diagnostics/allow-diagnostics'`

**Comment by pete-watters** [comment]
There's some valid test failures we will need to fix. I will help with that soon

**Comment by pete-watters** [comment]
I did some refactoring here and got the tests to pass. We should get another review before squashing commits then approving and merging.

File: src/app/ui/components/containers/dialog/dialog.tsx
Comment (kyranjamie) [inline src/app/ui/components/containers/dialog/dialog.tsx:17]: Not a change in this PR but, why not ReactNode here 🤔

**Review by kyranjamie — APPROVED** [review]
+1 to rebasing. @markmhendrickson when updating a PR, good practice to use `git commit --amend` flag to not create unnecessary commits.

Otherwise, LGTM. Love to see more lines of code removed than added.

File: src/app/ui/components/containers/dialog/dialog.tsx
Comment (pete-watters) [reply src/app/ui/components/containers/dialog/dialog.tsx:17]: I will change it separately👍 . Not sure why I didn't do that in the first place

**Pull request merged**
